packages/browser/__tests__/Session.spec.ts

```diff
@@ -162,17 +162,62 @@ describe("Session", () => {
   });
 
   describe("handleincomingRedirect", () => {
+    it("uses current window location as default redirect URL", async () => {
+      // We can't simply do 'window.location.href = defaultLocation;', as that
+      // causes our test environment to try to navigate to the new location (as
+      // a browser would do). So instead we need to reset 'window.location' as a
+      // whole, and reset it's value after our test. We also seem to need to
+      // save and restore the 'history' object.
+      const {
+        location,
+        history: { replaceState },
+      } = window;
+
+      try {
+        // location and history aren't optional on window, which makes TS complain
+        // (rightfully) when we delete them. However, they are deleted on purpose
+        // here just for testing.
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        delete window.location;
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        delete window.history.replaceState;
+
+        // Set our window's location to our test value.
+        const defaultLocation = "https://coolSite.com/myAppTestPage";
+        window.location = {
+          href: defaultLocation,
+        } as Location;
+        window.history.replaceState = jest.fn();
+
+        const clientAuthentication = mockClientAuthentication();
+        const clientAuthnHandler = jest.spyOn(
+          clientAuthentication,
+          "handleIncomingRedirect"
+        );
+
+        const mySession = new Session({ clientAuthentication });
+        await mySession.handleIncomingRedirect();
+        expect(clientAuthnHandler).toHaveBeenCalledWith(defaultLocation);
+      } finally {
+        // Restore our 'window' state.
+        window.location = location;
+        window.history.replaceState = replaceState;
+      }
+    });
+
     it("wraps up ClientAuthentication handleIncomingRedirect", async () => {
       // eslint-disable-next-line no-restricted-globals
       history.replaceState = jest.fn();
       const clientAuthentication = mockClientAuthentication();
-      const clientAuthnHandle = jest.spyOn(
+      const clientAuthnHandler = jest.spyOn(
```

Comment:
This was named `clientAuthnHandle` because it spies on the `handle` function, and isn't a "handler" per se. Why rename it ?

Reply:
Yeah, I was in two minds here - I don't really like the original name, but I didn't want to change it too much. I would say any `handleXXXX()` function is by definition as 'handler', and so this code is spying on a handler (and not a handle). Plus a 'handle' has other connotations. But overall maybe I'll just rename this `incomingRedirectHandler`, i.e., is the `clientAuthn` prefix really relevant here...?

Reply:
Yeah, I went with `incomingRedirectHandler`: 7f8d4af